client/session_container.go

```diff
@@ -0,0 +1,59 @@
+package client
+
+import (
+	v2session "github.com/nspcc-dev/neofs-api-go/v2/session"
+	"github.com/nspcc-dev/neofs-sdk-go/session"
+)
+
+// sessionContainer is a special type which unifies session logic management for client parameters.
+// All methods make public, because sessionContainer is included in Prm* structs.
+type sessionContainer struct {
+	isSessionIgnored bool
+	meta             v2session.RequestMetaHeader
+}
+
+// GetSession returns session object.
+//
+// Returns:
+//   - [ErrNoSession] err if session wasn't set.
+//   - [ErrNoSessionExplicitly] if IgnoreSession was used.
+func (x *sessionContainer) GetSession() (*session.Object, error) {
+	if x.isSessionIgnored {
+		return nil, ErrNoSessionExplicitly
+	}
+
+	token := x.meta.GetSessionToken()
+	if token == nil {
+		return nil, ErrNoSession
+	}
+
+	var sess session.Object
+	if err := sess.ReadFromV2(*token); err != nil {
+		return nil, err
+	}
+
+	return &sess, nil
+}
+
+// WithinSession specifies session within which the search query must be executed.
```

Comment:
is `search query` phrase appropriate for all methods accepting this prm?

Reply:
Not only search, for all queries. `search` remove from the comment